src/arch/arm/insts/static_inst.cc

```diff
@@ -303,7 +303,7 @@ ArmStaticInst::printIntReg(std::ostream &os, RegIndex reg_idx,
             ccprintf(os, "ureg0");
         else if (reg_idx == INTREG_SPX)
             ccprintf(os, "%s%s", (opWidth == 32) ? "w" : "", "sp");
-        else if (reg_idx == INTREG_X31)
+        else if (reg_idx == INTREG_X31 || reg_idx == INTREG_ZERO)
             ccprintf(os, "%szr", (opWidth == 32) ? "w" : "x");
```

Comment:
INTREG_ZERO may be created manually (not parsed) by gem5/arm to explicitly indicate "do not print anything":
src/arch/arm/isa/formats/data.isa:            dest = "INTREG_ZERO"
src/arch/arm/isa/formats/data.isa:            op1 = "INTREG_ZERO"
src/arch/arm/isa/formats/data.isa:            dest = "INTREG_ZERO"
src/arch/arm/isa/formats/data.isa:            op1 = "INTREG_ZERO"
src/arch/arm/isa/formats/data.isa:        "tst" : decInst("Tst", "INTREG_ZERO"),
src/arch/arm/isa/formats/data.isa:        "mov" : decInst("Mov", op1="INTREG_ZERO"),
src/arch/arm/isa/formats/data.isa:        "mvn" : decInst("Mvn", op1="INTREG_ZERO"),
src/arch/arm/isa/formats/data.isa:        "teq" : decInst("Teq", dest="INTREG_ZERO"),
src/arch/arm/isa/formats/data.isa:        "cmn" : decInst("Cmn", dest="INTREG_ZERO"),
src/arch/arm/isa/formats/data.isa:        "cmp" : decInst("Cmp", dest="INTREG_ZERO"),
src/arch/arm/isa/formats/data.isa:        "tst" : decInst("Tst", "INTREG_ZERO"),
src/arch/arm/isa/formats/data.isa:        "mov" : decInst("Mov", op1="INTREG_ZERO"),
src/arch/arm/isa/formats/data.isa:        "mvn" : decInst("Mvn", op1="INTREG_ZERO"),
src/arch/arm/isa/formats/data.isa:        "teq" : decInst("Teq", "INTREG_ZERO"),
src/arch/arm/isa/formats/data.isa:        "cmn" : decInst("Cmn", "INTREG_ZERO"),
src/arch/arm/isa/formats/data.isa:        "cmp" : decInst("Cmp", "INTREG_ZERO"),
In such case, this change looks wrong as it may break the arm32 disassembly.